### PR TITLE
chore: try to better handle misbehavior related to ZCFSeat.fail and ZCFSeat.exit

### PR DESCRIPTION
**Pull request by katelynsills**
If an `offerHandler` throws with `reason` undefined, fail with a new Error explaining that the contract code needs to be fixed.

If `fail` is called with a string, convert it to an Error, but make a note that it is incorrect.

### CODE REVIEW

**Review by Chris-Hibbert — APPROVED** [review]
LGTM

Should we fix the type declarations to not say that the Error is optional?

```
 * @param {Error=} reason
```

**Comment by katelynsills** [comment]
> Should we fix the type declarations to not say that the Error is optional?

I think that's a good idea. I can't think of what purpose not specifying the error would serve. I'll try making that change and see if anything breaks.

**Comment by erights** [comment]
Do you understand the one CI failure?

File: packages/zoe/src/contractFacet/zcfSeat.js [hunk 1]
```diff
@@ -144,6 +144,13 @@ export const createSeatManager = (zoeInstanceAdmin, getMathKindByBrand) => {
           'Seat exited with failure. Please check the log for more information.',
         ),
       ) => {
+        if (typeof reason === 'string') {
+          reason = Error(reason);
+          assert.note(
+            reason,
```
Comment (erights) [inline packages/zoe/src/contractFacet/zcfSeat.js:150]: Nice!

**Review by erights — APPROVED** [review]
LGTM

**Comment by katelynsills** [comment]
> Do you understand the one CI failure?

I think that's a treasury problem that is being fixed but is known

**Pull request merged**